holmes/plugins/toolsets/azuremonitor_metrics/azuremonitor_metrics_toolset.py

```diff
@@ -0,0 +1,1306 @@
+    cluster_name: Optional[str] = None
+    cluster_resource_id: Optional[str] = None
+    auto_detect_cluster: bool = True
+    tool_calls_return_data: bool = True
+    headers: Dict = {}
+    # Step size and data limiting configuration
+    default_step_seconds: int = 3600  # 1 hour default step size
+    min_step_seconds: int = 60       # Minimum 1 minute step size
+    max_data_points: int = 1000      # Maximum data points per query
+    # Internal fields for Azure authentication
+    _credential: Optional[Any] = None
+    _token_cache: Dict = {}
+
+    @field_validator("azure_monitor_workspace_endpoint")
+    def ensure_trailing_slash(cls, v: Optional[str]) -> Optional[str]:
+        if v is not None and not v.endswith("/"):
+            return v + "/"
+        return v
+
+    class Config:
+        arbitrary_types_allowed = True
+
+class BaseAzureMonitorMetricsTool(Tool):
+    """Base class for Azure Monitor Metrics tools."""
+    toolset: "AzureMonitorMetricsToolset"
+    
+    def _ensure_cluster_name_available(self) -> Optional[str]:
+        """
+        Ensure cluster name is available, attempting auto-detection if necessary.
+        
+        Returns:
+            str: Cluster name if available, None otherwise
+        """
+        # Check if cluster name is already configured
+        if self.toolset.config and self.toolset.config.cluster_name:
+            return self.toolset.config.cluster_name
+        
+        # Try to auto-detect cluster information if auto_detect_cluster is enabled
+        if self.toolset.config and self.toolset.config.auto_detect_cluster:
+            try:
+                logging.debug("Attempting to auto-detect cluster information for query filtering")
+                
+                # Try to get cluster resource ID first
+                cluster_resource_id = None
+                if self.toolset.config.cluster_resource_id:
+                    cluster_resource_id = self.toolset.config.cluster_resource_id
+                else:
+                    cluster_resource_id = get_aks_cluster_resource_id()
+                
+                if cluster_resource_id:
+                    # Extract cluster name from resource ID
+                    cluster_name = extract_cluster_name_from_resource_id(cluster_resource_id)
+                    
+                    if cluster_name:
+                        # Update the configuration with the detected values
+                        self.toolset.config.cluster_name = cluster_name
+                        self.toolset.config.cluster_resource_id = cluster_resource_id
+                        
+                        logging.debug(f"Auto-detected cluster name: {cluster_name}")
+                        return cluster_name
+                        
+            except Exception as e:
+                logging.debug(f"Failed to auto-detect cluster information: {e}")
+        
+        return None
+
+class CheckAKSClusterContext(BaseAzureMonitorMetricsTool):
```

Comment:
Can elaborate why need this tool? can we check it once when we load the tools? what is the use case for this tool?